File: JULY-2025/14-07-2025/p1290.scala
/**
 * Definition for singly-linked list.
 * class ListNode(_x: Int = 0, _next: ListNode = null) {
 *   var next: ListNode = _next
 *   var x: Int = _x
 * }
 */
object Solution {
    def getDecimalValue(head: ListNode, result: Int = 0): Int = {
        head match
            case null => result
            case _ => getDecimalValue(head.next, 2 * result + head.x)
    }
}

File: JULY-2025/14-07-2025/p1290.php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution {

    /**
     * @param ListNode $head
     * @return Integer
     */
    function getDecimalValue($head) {
         $current = $head;
         $binString ="";
        while ($current !== null) {
           $binString.= $current->val ;
            $current = $current->next;
        }
        return bindec($binString);
    }
}